content/guides/code-splitting.md

```diff
@@ -1,41 +1,199 @@
-
-Learn [how to split CSS](/guides/code-splitting-css) using the `ExtractTextWebpackPlugin`.
-
-
-## On Demand Code Splitting
-
-While resource splitting of the previous kind requires the user to specify the split points upfront in the configuration, one can also create dynamic split points in the application code.
-
-This can be used for more granular chunking of code, for example, per our application routes or as per predicted user behaviour. This allows the user to load non-essential assets on demand.
-
-Learn [how to split on demand](/guides/code-splitting-async) using `import()` or `require.ensure()`.
+module.exports = {
+  entry: {
+    index: './src/index.js',
+    vendor: [
+      'lodash'
+    ]
+  },
+  output: {
+    filename: '[name].bundle.js',
+    path: path.resolve(__dirname, 'dist')
+  }
+};
+```
+
+This will yield the following build result:
+
+``` bash
+Hash: d1e4eab63dc6ba09c930
+Version: webpack 2.6.1
+Time: 531ms
+           Asset    Size  Chunks                    Chunk Names
+vendor.bundle.js  544 kB       0  [emitted]  [big]  vendor
+ index.bundle.js  544 kB       1  [emitted]  [big]  index
+   [0] ./~/lodash/lodash.js 540 kB {0} {1} [built]
+   [1] (webpack)/buildin/global.js 509 bytes {0} {1} [built]
+   [2] (webpack)/buildin/module.js 517 bytes {0} {1} [built]
+   [3] ./src/index.js 216 bytes {1} [built]
+   [4] multi lodash 28 bytes {0} [built]
+```
+
+As mentioned there are some pitfalls to this approach:
+
+- If there are any duplicated modules between entry chunks they will be included in both bundles.
+- It isn't as flexible and can't be used to dynamically split code with the core application logic.
+
+The first of these two points is definitely an issue for our example, as `lodash` is also imported within `./src/index.js` and will thus be duplicated in the output. See the `CommonChunkPlugin` example below for a solution to this problem.
+
+
+## Plugins & Loaders
+
+There are multiple plugins and loaders in webpack's ecosystem that can help with splitting and managing split code. The most widely utilized of these is the [`CommonsChunkPlugin`](/plugins/commons-chunk-plugin), a fairly advanced tool that allows us to extract common dependencies into an existing entry chunk or an entirely new chunk. Let's use this to de-duplicate the `lodash` dependency from the previous example:
+
+__webpack.config.js__
+
+``` js
+const path = require('path');
+const webpack = require('webpack');
+
+module.exports = {
+  entry: {
+    index: './src/index.js',
+    vendor: [
+      'lodash'
+    ]
+  },
+  plugins: [
+    new webpack.optimize.CommonsChunkPlugin({
+      name: 'vendor' // Specify the common bundle's name.
+    })
+  ],
+  output: {
+    filename: '[name].bundle.js',
+    path: path.resolve(__dirname, 'dist')
+  }
+};
+```
```

Comment:
> I'm not convinced this is the correct order to teach it. It might just be me, but I wanted `CommonsChunkPlugin` to extract all common modules from entry points into a new file.

I tweaked the example so it demonstrates exactly what you describe as I agree that it's a more intuitive way to teach it. There are now two separate `src` modules that duplicate the `lodash` dependency and the _Prevent Duplication_ section resolves that duplication.